tracer/src/Datadog.Trace/Activity/ActivityListener.cs

```diff
@@ -0,0 +1,295 @@
+                    return;
+                }
+
+                CreateCurrentActivityDelegates(activityType);
+                ChangeActivityDefaultFormat(activityType);
+
+                if (version.Major >= 5)
+                {
+                    // if Version >= 5 loaded (Uses ActivityListener implementation)
+                    CreateActivityListenerInstance(activityType);
+                }
+                else
+                {
+                    // if Version is 4.0.4 or greater (Uses DiagnosticListener implementation / Nuget version 4.6.0)
+                    CreateDiagnosticSourceListenerInstance(diagnosticListenerType);
+                }
+
+                return;
+            }
+
+            Log.Information("An activity listener was found but version {version} is not supported.", version?.ToString() ?? "(null)");
+
+            static void CreateCurrentActivityDelegates(Type activityType)
+            {
+                // Create Activity.Current delegate.
+                var activityCurrentProperty = activityType.GetProperty("Current", BindingFlags.Public | BindingFlags.Static);
+                if (activityCurrentProperty is null)
+                {
+                    throw new NullReferenceException("Activity.Current property cannot be found in the Activity type.");
+                }
+
+                var activityCurrentMethodInfo = activityCurrentProperty.GetMethod!;
+                var activityCurrentDynMethod = new DynamicMethod("ActivityCurrent", typeof(object), Type.EmptyTypes, typeof(ActivityListener).Module, true);
+                var activityCurrentDynMethodIl = activityCurrentDynMethod.GetILGenerator();
+                activityCurrentDynMethodIl.EmitCall(OpCodes.Call, activityCurrentMethodInfo, null);
+                activityCurrentDynMethodIl.Emit(OpCodes.Ret);
+                _getCurrentActivity = (Func<object>)activityCurrentDynMethod.CreateDelegate(typeof(Func<object>));
+            }
+
+            static void ChangeActivityDefaultFormat(Type activityType)
+            {
+                // We change the default ID format to W3C (so traceid and spanid gets populated)
+                if (Activator.CreateInstance(activityType, string.Empty).TryDuckCast<IActivityFormat>(out var activityFormat))
+                {
+                    activityFormat.DefaultIdFormat = ActivityIdFormat.W3C;
+                }
+            }
+        }
+
+        public static void StopListeners()
+        {
+            // If there's an activity listener instance we detach the instance and clear it.
+            if (_activityListenerInstance is { } activityListenerInstance)
+            {
+                var activitySourceType = Type.GetType("System.Diagnostics.ActivitySource, System.Diagnostics.DiagnosticSource", throwOnError: true);
+                var detachListenerMethodInfo = activitySourceType!.GetMethod("DetachListener", BindingFlags.Static | BindingFlags.Public);
```

Comment:
I think null propagating instead of dammit makes sense?
```suggestion
                var detachListenerMethodInfo = activitySourceType?.GetMethod("DetachListener", BindingFlags.Static | BindingFlags.Public);
```

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/pull/2446/commits/8533973cae079282f05839fc49fad6d16a526b83